iina/MPVPlaylistItem+Scripting.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  MPVPlaylistItem+Scripting.swift
+//  iina
+//
+//  Created by Nate Weaver on 2020-03-07.
+//  Copyright © 2020 lhc. All rights reserved.
+//
+
+import Foundation
+
+extension MPVPlaylistItem {
+
+  @objc override var classCode: FourCharCode {
+    return "cPlL"
+  }
+
+  var scriptingIndex: Int? {
+    return player?.info.playlist.firstIndex { $0 === self }
+  }
+
+  @objc override var objectSpecifier: NSScriptObjectSpecifier? {
+    guard let player = player else { return nil }
+
+    let containerClass = NSScriptClassDescription(for: PlayerCore.self);
+    let containerSpecifier = player.objectSpecifier
+
+    guard let index = scriptingIndex else { return nil }
+
+    return NSIndexSpecifier(containerClassDescription: containerClass!, containerSpecifier: containerSpecifier, key: "scriptingPlaylistItems", index: index)
+  }
+
+}
+
+extension MPVPlaylistItem {
+
+  @objc var scriptingURL: String { isNetworkResource ? filename : URL(fileURLWithPath: filename).absoluteString }
```

Comment:
I think naming this "URL" might a bit confusing if it's not actually a URL.

Reply:
Good call; I’ll change it to `scriptingURLString` or something.